api/lib/project-fields-nodes-to-fields-map.js

```diff
@@ -58,12 +65,18 @@ export function projectFieldsNodesToFieldsMap(state, project, nodes) {
       );
 
       if (!node) {
-        const projectFieldNames = nodes.map((node) => `"${node.name}"`);
-        throw new Error(
-          `[github-project] "${userFieldName}" could not be matched with any of the existing field names: ${projectFieldNames.join(
-            ", "
-          )}`
+        const projectFieldNames = nodes
+          .map((node) => `"${node.name}"`)
+          .join(", ");
+        if (!fieldOptional) {
+          throw new Error(
+            `[github-project] "${userFieldName}" could not be matched with any of the existing field names: ${projectFieldNames}`
```

Comment:
I'd include instructions how to disable this error by making the field optional. Something like this

```suggestion
            `[github-project] "${userFieldName}" could not be matched with any of the existing field names: ${projectFieldNames}`. If the field should be considered optional, then set it to "${userInternalFieldName}: { name: "${userFieldName}", optional: true}"
```